server/src/main/java/message/MessagePosterImpl.java

```diff
@@ -0,0 +1,65 @@
+package message;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.Timestamp;
+import com.google.cloud.firestore.*;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import responses.NewPostResponse;
+
+
+public class MessagePosterImpl implements MessagePoster {
+
+    @VisibleForTesting
+    static final String COLLECTION_PATH = "messages";
```

Comment:
Nit: maybe just move this into Constants as well since it's duplicated in MessageFinderIMpl?

Reply:
sounds good